main.rb

```diff
@@ -0,0 +1,110 @@
+# rubocop:disable Style/CaseEquality
+
+module Enumerable
+  def my_each
+    size.times { |counter| yield(to_a[counter]) } if block_given?
+    block_given? ? self : enum_for(__method__)
+  end
+
+  def my_each_with_index
+    size.times { |counter| yield(to_a[counter], counter) } if block_given?
+    block_given? ? self : enum_for(__method__)
+  end
+
+  def my_select
+    return enum_for(__method__) unless block_given?
+
+    selected = []
+    my_each { |item| selected.push(item) if yield(item) }
+    selected
+  end
+
+  def my_all?(*args, &block)
+    all_true = true
+
+    if (arg = args[0])
+      puts 'Warning, the block is being ignored.' if block
+
+      my_each do |item|
+        all_true &= if arg.is_a? Regexp
+                      (arg === item.to_s)
+                    else
+                      item.is_a? arg
+                    end
+      end
+    elsif block_given?
+      my_each { |item| all_true &= yield(item) }
+    else
+      all_true = my_all? { |item| item }
+    end
+
+    all_true
+  end
+
+  def my_any?(*args, &block)
+    result = false
+    if (arg = args[0])
+      puts 'Warning, the block is being ignored. ' if block
+
+      my_each { |item| result |= (arg.is_a?(Regexp) ? arg === item.to_s : item.is_a?(arg)); }
+    elsif block_given?
+      my_each { |item| result |= yield(item) }
+    else
+      return my_any? { |item| item }
+    end
+    result
+  end
```

Comment:
### #my_any?
When testing this method, each time that I give it an argument I received the following` TypeError: Class or module required.`
here I'll add an example of one of the tests:
``` ruby
irb> arr = %w[dog door rod blade] 
irb> arr.my_any?('cat')
  => TypeError (class or module required)
```
> expected to be false

Reply:
Fixed